### PR TITLE
Feat/custom trigger priority

### PR DESCRIPTION
**Pull request by wysohn**

### CODE REVIEW

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitEventRegistry.java [hunk 1]
```diff
@@ -98,17 +100,34 @@ public Class<?> getEvent(String eventStr) throws ClassNotFoundException {
         return event;
     }
 
+    private EventPriority getPriority(int priority) {
```
Comment (Sayakie) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitEventRegistry.java:103]: Maybe we should change the visibility modifier `private` → `public`, but just my opinion

File: core/src/main/java/io/github/wysohn/triggerreactor/core/IEventHook.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package io.github.wysohn.triggerreactor.core;
 
-@FunctionalInterface
 public interface IEventHook {
+    int getPriority();
```
Comment (Sayakie) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/IEventHook.java:4]:
```suggestion
    default int getPriority() {
        return -1;
    }
```
Add `default` access modifier so we can keep using `FunctionalInterface`

**Review by Sayakie — CHANGES_REQUESTED** [review]

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitEventRegistry.java (hunk 1 above)
Comment (wysohn) [reply bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitEventRegistry.java:103]: > Maybe we should change the visibility modifier `private` → `public`, but just my opinion

Sounds legit, but this is merely to map the priority value to the `EventPriority`, so there wouldn't be much usecase for it other than the BukkitEventRegistry class itself

Lets comeback to it and fix it if we really need it

Comment (Sayakie) [reply bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitEventRegistry.java:103]: Ahh, that makes sense a lot.

File: core/src/main/java/io/github/wysohn/triggerreactor/core/IEventHook.java [hunk 3]
```diff
@@ -1,6 +1,19 @@
 package io.github.wysohn.triggerreactor.core;
 
-@FunctionalInterface
 public interface IEventHook {
```
Comment (Sayakie) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/IEventHook.java:3]:
```suggestion
@FunctionalInterface
public interface IEventHook {
```

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/custom/CustomTrigger.java [hunk 4]
```diff
@@ -87,6 +88,20 @@ public String getEventName() {
         return eventName;
     }
 
+    public int getPriority() {
```
Comment (Sayakie) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/custom/CustomTrigger.java:91]:
```suggestion
    @Override
    public int getPriority() {
```

**Review by Sayakie — CHANGES_REQUESTED** [review]
Random annotations are missed.

**Comment by wysohn** [comment]
@Sayakie 
I didn't know that I could make the commit suggestion right away! That's very cool

**Review by Sayakie — APPROVED** [review]
LGTM!

**Pull request merged**